lib/intercept.js

```diff
@@ -4,10 +4,9 @@
  * @module nock/intercepts
  */
 
-const { InterceptedRequestHandler } = require('./request_overrider')
+const { InterceptedRequestBroker } = require('./request_overrider')
```

Comment:
I'm thinking `InterceptedRequestRouter` might be a better name for this first class.